Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -257,6 +259,34 @@ - (void)viewDidLoad {
     self.webView.scrollView.layer.anchorPoint = CGPointMake((isRTL ? 1.0 : 0.0), 0.0);
 
     [self tocUpdateViewLayout];
+
+    [self.KVOControllerNonRetaining observe:self.webView.scrollView keyPath:WMF_SAFE_KEYPATH(UIScrollView.new, contentSize) options:NSKeyValueObservingOptionNew block:^(WebViewController* observer, id object, NSDictionary* change) {
+        [self.sectionTitlesViewController updateOverlaysPositions];
+        // Restrict the web view from scrolling horizonally.
+        [object preventHorizontalScrolling];
+    }];
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(editSection:)
+                                                 name:@"EditSection"
+                                               object:nil];
+}
+
+- (void)editSection:(NSNotification*)notification {
+    if ([self tocDrawerIsOpen]) {
+        [self tocHide];
+        return;
+    }
+
+    if (self.editable) {
+        WMFTitleOverlayLabel* sender = notification.object;
+        self.sectionToEditId = [sender sectionId];
+        [self showSectionEditor];
```

Comment:
Instead of setting an ivar and then calling a method with no parameters, you can do something like:

```
- (void)showSectionEditorForSection:(NSUInteger)sectionID
```

Then you can get rid of the ivar… similar to another comment I left earlier.